spring-vault-core/src/main/java/org/springframework/vault/authentication/AppRoleAuthentication.java

```diff
@@ -105,7 +106,7 @@ public static AuthenticationSteps createAuthenticationSteps(
 		if ((roleId instanceof Wrapped || roleId instanceof Pull)
 				&& (secretId instanceof Wrapped || secretId instanceof Pull)) {
 
-			throw new IllegalArgumentException(
+			throw new VaultClientException(
```

Comment:
This was one of the in between ones I guess could go either way. My only argument is hey if ones unifying up the hierarchy why ISN't this a VCE.

While I use IllegalState, IllegalArgument a lot, in an API, I'd argue unless it is a 100% obvious and expected consequence, it's best shielded, otherwise encouraging people to just catch RuntimeException.

But continuing on...

Reply:
We're using IAE and ISE to distinguish between framework-specific issues and state/argument issues. There's not point in introducing own exceptions of a generic type that shadow the fact that IAE and ISE are usually programming bugs while `VaultException`s might be something to recover from.